src/app/services/course-enrollment.service.ts

```diff
@@ -0,0 +1,24 @@
+import { Injectable } from '@angular/core';
+import {inject} from '@angular/core'
+import { HttpClient } from '@angular/common/http';
+import { environment } from '../../enviroments/enviroment';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class CourseEnrollmentService {
+  #http = inject(HttpClient)
+  // #token = inject(TokenService)
+
+  enrollCourse (courseId: string) {
+    // Use token
+    return this.#http.put(`${environment.API_URL}/api/course/enroll/${courseId}`, {})
+  }
+
+  unrollCourse (courseId: string) {
+    // Use token
```

Comment:
Make an interceptor to inject the token, so u don't have to manually inject it on every request

Reply:
Ok, I'll be fixing this and then I merge